HLTrigger/special/src/HLTRegionalEcalResonanceFilter.cc

```diff
@@ -1022,25 +1027,35 @@ void HLTRegionalEcalResonanceFilter::calcShowerShape(const reco::BasicCluster &b
 
 void HLTRegionalEcalResonanceFilter::calcPaircluster(const reco::BasicCluster &bc1, const reco::BasicCluster &bc2,float &m_pair,float &pt_pair,float &eta_pair, float &phi_pair){
     
-  
-  float theta1 = 2. * atan(exp(-bc1.eta())); 
+  // use XYZVector instead of TLorentzVector to make things faster (and initialize with cartesian coordinates).
+  // We are interested in the momentum vector:  however, we start from cartesian coordinates to get the vector direction, 
+  // then we set the vector's magnitude to obtain momentum coordinates. The magnitude we set is equal to the particle's energy.
+  // We can do this because, assuming massless particles (or negligible mass), the magnitude of the momentum vector is given by the energy.
+
+  math::XYZVector v1( bc1.x(), bc1.y(), bc1.z() ); 
   float en1 = bc1.energy();
-  float pt1 = en1 * sin(theta1); 
-  TLorentzVector v1( pt1 *cos(bc1.phi()),pt1 * sin(bc1.phi()),en1*cos(theta1),en1);
-  
-  float theta2 = 2. * atan(exp(-bc2.eta())); 
-  float en2 = bc2.energy();
-  float pt2 = en2 * sin(theta2); 
-  TLorentzVector v2( pt2 *cos(bc2.phi()),pt2 * sin(bc2.phi()),en2*cos(theta2),en2);
-  
-  TLorentzVector v = v1 + v2; 
-  
-  m_pair = v.M();
-  pt_pair = v.Pt();
-  eta_pair = v.Eta();
-  phi_pair = v.Phi();
-  
-  
+  float scaleFactor = en1 / v1.R();  // XYZVector::R() returns sqrt(Mag2()), where Mag2()= fx*fx + fy*fy + fz*fz 
+  // here I'm assuming that the vector initial magnitude is always different from 0 (the cluster must be somewhere in the detector, so the distance is greater than 0)
+  v1 *= scaleFactor;
+
+  // vsum would be v1 + v2, but instead of declaring both v2 and vsum, just declare vsum, initialize as if it is v2 and then sum v1.
+  math::XYZVector vsum(bc2.x(), bc2.y(), bc2.z());
```

Comment:
You're right. I missed that. Will fix soon. 

Inviato da iPhone

> Il giorno 06 mag 2016, alle ore 18:57, Andrea Bocci <notifications@github.com> ha scritto:
> 
> In HLTrigger/special/src/HLTRegionalEcalResonanceFilter.cc:
> 
> > -  TLorentzVector v2( pt2 *cos(bc2.phi()),pt2 * sin(bc2.phi()),en2*cos(theta2),en2);
> > -  
> > -  TLorentzVector v = v1 + v2; 
> > -  
> > -  m_pair = v.M();
> > -  pt_pair = v.Pt();
> > -  eta_pair = v.Eta();
> > -  phi_pair = v.Phi();
> > -  
> > -  
> > +  float scaleFactor = en1 / v1.R();  // XYZVector::R() returns sqrt(Mag2()), where Mag2()= fx*fx + fy*fy + fz*fz 
> > +  // here I'm assuming that the vector initial magnitude is always different from 0 (the cluster must be somewhere in the detector, so the distance is greater than 0)
> > +  v1 *= scaleFactor;
> > +
> > +  // vsum would be v1 + v2, but instead of declaring both v2 and vsum, just declare vsum, initialize as if it is v2 and then sum v1.
> > +  math::XYZVector vsum(bc2.x(), bc2.y(), bc2.z());
> same here
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>